tests/integration/features/search.feature

```diff
@@ -0,0 +1,259 @@
+    Then the card "Overdue task" is not found
+    Then the card "Future task" is found
+    Then the card "Tomorrow task" is not found
+    Then the card "Next week task" is not found
+
+  Scenario: Search for assigned user
+    Given user "user1" exists
+    And shares the board with user "user1"
+    Given create a card named "Assigned card to user1"
+    And assign the card to the user "user1"
+    When searching for 'assigned:user1'
+    Then the card "Example task 1" is not found
+    And the card "Assigned card to user1" is found
+
+  Scenario: Search for assigned user by displayname
+    Given user "ada" with displayname "Ada Lovelace" exists
+    And shares the board with user "ada"
+    Given create a card named "Assigned card to ada"
+    And assign the card to the user "ada"
+    When searching for 'assigned:"Ada Lovelace"'
+    Then the card "Example task 1" is not found
+    And the card "Assigned card to ada" is found
+
+  Scenario: Search for assigned users
+    Given user "user1" exists
+    And shares the board with user "user1"
+    Given create a card named "Assigned card to user0"
+    And assign the card to the user "user0"
+    Given create a card named "Assigned card to user01"
+    And assign the card to the user "user0"
+    And assign the card to the user "user1"
+    When searching for 'assigned:user0 assigned:user1'
+    Then the card "Example task 1" is not found
+    And the card "Assigned card to user0" is not found
+    And the card "Assigned card to user01" is found
+
+  Scenario: Search for assigned group
+    Given user "user1" exists
+    And shares the board with user "user1"
+    Given group "group1" exists
+    And shares the board with group "group1"
+    Given user "user1" belongs to group "group1"
+    Given create a card named "Assigned card to group1"
+    And assign the card to the group "group1"
+    When searching for 'assigned:user1'
+    Then the card "Example task 1" is not found
+    And the card "Assigned card to group1" is found
+
+    When searching for 'assigned:group1'
+    Then the card "Example task 1" is not found
+    And the card "Assigned card to group1" is found
+
+  Scenario: Search for assigned tag
+    Given create a card named "Labeled card"
+    # Default labels from boards are used for this test case
+    And assign the tag "Finished" to the card
+    When searching for 'tag:Finished'
+    Then the card "Example task 1" is not found
+    And the card "Labeled card" is found
+
+    Given create a card named "Multi labeled card"
+    And assign the tag "Finished" to the card
+    And assign the tag "To review" to the card
+    When searching for 'tag:Finished tag:Later'
+    Then the card "Example task 1" is not found
+    And the card "Multi labeled card" is not found
+
+    When searching for 'tag:Finished tag:"To review"'
+    Then the card "Example task 1" is not found
+    And the card "Labeled card" is not found
+    And the card "Multi labeled card" is found
```

Comment:
also add tests for invalid expressions ? (if not covered by unit tests)

Reply:
There are not really invalid expressions since everything that is not being parsed is added as a simple text token for searching in the plain text fields of the card, but I'll cover the parser with some additional unit tests.

Reply:
Added some testing of to the parser be2f070